File: PYTHON/Numeric-Functions/numeric_bases_functions/decimal_to_hexadecimal.py
def decimal_to_hexadecimal (num):
    num = int(num)
    strr = str("")
    res = str("")
    while (num > 0):
        tempI = int(num / 16)
        tempF = float(num / 16.0)
        tempF = float(tempF - tempI)
        tempF = int(tempF * 16)
        if (tempF >= 0 and tempF <= 9):
            strr += str(tempF)
        else:
            tempF = int(tempF + 55)
            strr += chr(tempF)
        #print(tempF)
        num = int(tempI)
    lim = len(strr)
    for cnt in range(lim-1, -1, -1):
        res += str(strr[cnt])
    return res

x = int(input("User, enter a DECIMAL number: "))
results = decimal_to_hexadecimal(x)
print(str(x) + " --> " + str(results)) 

File: C#/NOT_WORKING/decimal_to_hexadecimal.cs
namespace aviv_yunker {
	class decimalToHexadecimal {
		public static void Main (string [] args) {
			string x, results;
			int num;
			System.Console.WriteLine("User, enter a number: ");
			x = System.Console.ReadLine();
			num = int.Parse(x);
			results = decimal_to_hexadecimal(x);
			System.Console.WriteLine("DEC({0}) --> HEX({1})", x, results);
		}
		public static string decimal_to_hexadecimal (int num) {
			string tmp, res;
			int tempI;
			float tempF;
			while (num > 0) {
				tempI = (int)(num / 2);
				tempF = (float)(num / 2.0);
				tempF = (float)(2 * (tempF - tempI));
				if ((int)tempF >= 0 && (int)tempF <= 9) {
					tmp += tempF;
				}
				else if ((int)tempF >= 10 && (int)tempF <= 16) {
					tmp += (chr)(tempF+55);
				}
				num = tempI;
			}
			for (int i = tmp.length-1; i >= 0; i--) {
				res += tmp[i];
			}
			return res;
		}
	}
}
